### DISCUSSION TITLE
[FR] Keep delayed jobs when clearing queue within CP

### DISCUSSION THREAD

**Discussion by naboo** [craftcms/cms#9033] [Ideas]
### Description
When you clear the queue from within the CP (maybe because a task is stuck, or the que has been jammed because of something else) it would be ideal if not _all_ jobs are deleted. Future (delayed) jobs should not be deleted in my opinion.

**Comment by brandonkelly** [comment]
Hmmm, not sure about this one. I’d expect all jobs to be cleared when you tell it to clear all jobs.

**Comment by naboo** [comment]
Could it be so that there are two options?

I mean delayed jobs are yet to run. They haven't "done anything" yet which
current running ones might have (that needs clearing).

I pressed the clear queue button today on a site which had +20 delayed jobs
and it wiped everything. The delayed jobs were things to be executed later
in the week.

Den tis 4 maj 2021 16:58Brandon Kelly ***@***.***> skrev:

**Comment by brandonkelly** [comment]
> I mean delayed jobs are yet to run. They haven't "done anything" yet

Nor have _any_ jobs besides the one job currently being executed (if there is one).

**Comment by naboo** [comment]
That is true ofc.

Well if you would mind giving it some thought that be appreciated.

If you aren't aware you might (as I did) clear an entire list of future
jobs :)

Maybe an alert to what is going to happen at least?

Den tis 4 maj 2021 18:28Brandon Kelly ***@***.***> skrev:

**Comment by brandonkelly** [comment]
> The delayed jobs were things to be executed later in the week.

You might want to give [omnilight/yii2-scheduling](https://github.com/omnilight/yii2-scheduling) a look – it’s a library that lets you schedule console controller actions to be executed at certain intervals, which sounds like it might be more appropriate than the queue for your use case.

We’re using it for various things internally, and it’s pretty easy to hook up to a Craft project. 

- https://github.com/pixelandtonic/craftnet/blob/master/config/app.php#L91-L94
- https://github.com/pixelandtonic/craftnet/blob/5b552f059a9ee19911b7b644b7d5b3639b4caa58/.ebextensions/09_cron.config#L32
- https://github.com/pixelandtonic/craftnet/blob/master/config/schedule.php

**Comment by naboo** [comment]
Thanks for the tip Brandon. I'll look into it.
